app/lib/shared/exceptions.dart

```diff
@@ -157,14 +158,28 @@ class InvalidInputException extends ResponseException {
     _check(_ulidPattern.hasMatch(value), () => '"$name" is not a valid ulid.');
   }
 
-  static void checkSemanticVersion(String? version) {
+  /// Throw [InvalidInputException] if [package] is not an allowed package name.
+  static void checkPackageName(String? package) {
+    // TODO: Reuse logic from validatePackageName in pub_package_reader.dart
+    //       This is for new existing packages! Not only new packages!
+    checkNotNull(package, 'package');
+    checkStringLength(package!, 'package', minimum: 1, maximum: 64);
+    checkMatchPattern(package, 'package', RegExp(r'^[a-zA-Z0-9_]+$'));
+    checkMatchPattern(package, 'package', RegExp(r'^[a-zA-Z_]'));
```

Comment:
I think this is a bit confusing compared to one line up, maybe a single regexp that handles both the beginning and the rest of the package name would be better.